COMPILING.md

```diff
@@ -380,6 +380,47 @@ For the packaged builds of CBMC on our release page we currently build CBMC
 with the MiniSat2 SAT solver statically linked at compile time. However it is
 also possible to build CBMC using alternative SAT solvers.
 
+### Compiling with CaDiCaL
+
+The [CaDiCaL](https://github.com/arminbiere/cadical) solver supports the
+[IPASIR](https://github.com/biotomas/ipasir) C interface to incremental SAT
+solvers, which is also supported by CBMC. So the process for producing a CBMC
+with CaDiCaL build is to build CaDiCaL as a static library then compile CBMC
+with the IPASIR build options and link to the CaDiCaL static library.
```

Comment:
While that is one way of doing it, it's also possible to use the simpler approach using `-Dsat_impl=cadical` (with `cmake`) or `make -C src cadical-download` followed by `make -C src CADICAL=../../cadical`.

Reply:
Yes. This was to document how to do it with the IPASIR interface.

Reply:
I figured as much from the commit message, but could you perhaps also include a few words on those simpler approaches? I don't think the remainder of the section makes clear that this is only needed when choosing the IPASIR approach. (That said, it may be worth pondering long term whether we should stop supporting interfaces other than IPASIR to simplify maintenance.)

Reply:
Pushed a change now that I believe clarifies how to build and link different SAT solvers and makes clear the differences.